src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -1918,5 +1942,42 @@ void M(delegate*<void> ptr)
                 Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "ToString").WithArguments("delegate*<void>", "ToString").WithLocation(6, 13)
             );
         }
+
+        [Fact]
+        public void ArglistCannotBeUsedWithFunctionPointers()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe class C
+{
+    static void M(delegate*<string, int, void> ptr1, delegate*<__arglist, void> ptr2)
+    {
+        ptr1(__arglist(string.Empty, 1), 1);
+        ptr1(null, __arglist(string.Empty, 1));
+        ptr1(null, 1, __arglist(string.Empty, 1));
+        ptr2(__arglist(1, 2, 3, ptr1));
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (4,64): error CS1031: Type expected
+                //     static void M(delegate*<string, int, void> ptr1, delegate*<__arglist, void> ptr2)
```

Comment:
>__arglist [](start = 82, length = 9)

It would be good to assert value of MethodSymbol.IsVararg for this function pointer signature.

Reply:
Also how many parameters we have and what the types are.

---
In reply to: [413314469](https://github.com/dotnet/roslyn/pull/43355#discussion_r413314469) [](ancestors = 413314469)